projects/core/components/svg/svg.component.ts

```diff
@@ -62,39 +62,53 @@ export class TuiSvgComponent {
         @Inject(DomSanitizer) private readonly sanitizer: DomSanitizer,
         @Inject(ElementRef) private readonly elementRef: ElementRef<Element>,
         @Inject(TUI_SVG_SRC_PROCESSOR)
-        private readonly srcProcessor: TuiStringHandler<string>,
+        private readonly srcProcessor: TuiInjectionTokenType<
+            typeof TUI_SVG_SRC_PROCESSOR
+        >,
         @Inject(TUI_SVG_CONTENT_PROCESSOR)
-        private readonly contentProcessor: TuiStringHandler<string>,
+        private readonly contentProcessor: TuiInjectionTokenType<
+            typeof TUI_SVG_CONTENT_PROCESSOR
+        >,
     ) {
         this.innerHTML$ = this.src$.pipe(
-            switchMap(() =>
-                this.isExternal
-                    ? this.getExternalIcon(this.icon)
-                    : of(this.getSafeHtml(this.icon)),
-            ),
+            switchMap(() => {
+                if (typeof this.icon === `string`) {
+                    return this.isExternal
+                        ? this.getExternalIcon(this.icon)
+                        : of(this.getSafeHtml(this.icon));
+                }
+
+                return of(this.icon);
+            }),
             startWith(``),
         );
     }
 
     @Input()
     @tuiRequiredSetter()
-    set src(src: string) {
+    set src(src: SafeHtml | string) {
         this.icon = this.srcProcessor(src);
         this.src$.next();
     }
 
-    get src(): string {
+    get src(): SafeHtml | string {
         return this.icon;
     }
 
     get use(): string {
-        return this.icon.includes(`.svg#`)
-            ? this.icon
-            : this.resolveName(this.icon, this.iconsPath);
+        if (typeof this.icon === `string`) {
+            return this.icon.includes(`.svg#`)
+                ? this.icon
+                : this.resolveName(this.icon, this.iconsPath);
+        }
+
+        return ``;
     }
 
     get isInnerHTML(): boolean {
-        return this.isSrc || this.isExternal || (this.isName && this.isShadowDOM);
+        return typeof this.icon === `string`
```

Comment:
`return !(typeof this.icon === `string`) || this.isSrc || this.isExternal || (this.isName && this.isShadowDOM)`

Reply:
fixed